IndicoV2/Client/Submissions/ISubmissionsClient.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using IndicoV2.Client.Submissions.Models;
+
+namespace IndicoV2.Client.Submissions
+{
+    public interface ISubmissionsClient
+    {
+        Task<IEnumerable<int>> CreateAsync(int workflowId, Stream[] streams, CancellationToken cancellationToken = default);
+        Task<IEnumerable<int>> CreateAsync(int workflowId, Uri[] streams, CancellationToken cancellationToken = default);
+        Task<Job> GetJobAsync(int submissionId, CancellationToken cancellationToken = default);
```

Comment:
This approach seems to more user friendly than separate classes with properties and exec methods. It is also easier to provide unit tests for the code and also business objects are grouped together so it increase readability.